api/src/test/java/org/openmrs/PersonTest.java

```diff
@@ -723,6 +723,113 @@ public void getBirthDateTime_shouldReturnNullIfBirthtimeIsNull() throws Exceptio
 		Assert.assertNull(person.getBirthDateTime());
 	}
 
+	/**
+	 * @see Person#getAttribute(String)
+	 * @verifies get attribute based on String attributename
+	 */
+	@Test
+	public void  getAttribute_shouldPersonAttributeBasedOnAttributeName() throws Exception {
+		Person person = personHelper(false, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+		Assert.assertEquals("name3", person.getAttribute("name3").getAttributeType().getName());
+	}
+
+	/**
+	 * @see Person#getAttribute(String)
+	 * @verifies null if all voided, get attribute based on String attributename
+	 */
+	@Test
+	public void  getAttribute_shouldReturnNullIfAttributeNameIsVoided() throws Exception {
+		Person person = personHelper(true, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+		Assert.assertEquals(null, person.getAttribute("name3"));
+	}
+
+	/**
+	 * @see Person#getAttribute(PersonAttributeType)
+	 * @verifies null if all voided
+	 */
+	@Test
+	public void  getAttribute_shouldReturnNullWhenExistingPersonAttributeTypeIsVoided () throws Exception {
+		Person person = personHelper(true, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	 	PersonAttributeType type = new PersonAttributeType(new Integer(3));
+	 	type.setName("name3");
+		Assert.assertNull(null, person.getAttribute(type));
```

Comment:
This is incorrect, it should be

`Assert.assertNull(person.getAttribute(type));`